### DISCUSSION TITLE
Is it legal to submit the same PR to both Redis and Valkey?

### DISCUSSION THREAD

**Discussion by Nugine** [valkey-io/valkey#1286] [Q&A] [answered]
Recently I submitted a performance improvement PR to Redis (https://github.com/redis/redis/pull/13558).
I'm not sure whether the licenses allow me to send the same patch to Valkey, especially there are some changes made by code reviewers.
What should I do if I would like to port the performance improvement?

**Comment by madolson** [comment, marked as answer]
(Note, not a lawyer) You are the one with the copyright for your original work. The CLA you signed with Redis gives them the right to redistribute your work under their proprietary license. You are allowed to submit your original code to both Valkey and Redis, since you own it. If anyone that is not you made any changes before applying it, you are not allowed to submit that code since it's not under a license we can use.

For legal reasons I've been advised to not open the link you've sent me, but you should be OK to submit PR with the *original* content of the code you proposed to Redis along with any original changes you made.

If you want to avoid this type of confusion in the future, it's cleanest to contribute to Valkey first and then to Redis.
